src/utils/index.ts

```diff
@@ -50,20 +50,27 @@ export function areSameAddress(a: string, b: string) {
   return a.toUpperCase() === b.toUpperCase();
 }
 
-export function isValidBytes32CompliantString(value: string, variableName: string) {
-  if (value.length > 32) {
+/**
+ * Check the length of a given string to ensure it meets correct bounds
+ * @param value - the string itself
+ * @param variableName - the name of the variable associated to the string itself
+ * @param opts - optional min and max length of the string. Defaults to a minimum of 0 (empty string) and a maximum of 32 characters
+ */
+export function checkStringLength(
+  value: string,
+  variableName: string,
+  opts?: { minLength: number | undefined; maxLength: number }
```

Comment:
Can be rewritten as `opts?: { minLength?: number; maxLength: number } = { maxLength: 32 }`

And then in the function body you can just do

`const { minLength = 0, maxLength } = args;`

Also this means that when checking if minLength wasn't supplied you don't have to check if `opts` is truthy, since it has a default value

Reply:
"Function cannot have a question mark and an initializer"

`opts?: { minLength?: number; maxLength: number } = { maxLength: 32 }`

Reply:
And also, I'm not sure what to do with "args" that isn't present here
```
const { minLength = 0, maxLength } = args;
```